src/skarab_fpga.py

```diff
@@ -285,6 +289,56 @@ def read(self, device_name, size, offset=0):
         # return the number of bytes requested
         return data[offset: offset + size]
 
+    def bulk_read(self, device_name, size, offset=0):
+        """
+        Return size_bytes of binary data with carriage-return escape-sequenced.
+        :param device_name: name of memory device from which to read
+        :param size: how many bytes to read
+        :param offset: start at this offset, offset in bytes
+        :return: binary data string
+        """
+
+        # map device name to address, if can't find, bail
+        if device_name in self.memory_devices.keys():
```

Comment:
The default behaviour for a dict is to use keys() when you use 'in'. So 'if device_name in self.memory_devices' is the same as 'if device_name in self.memory_devices.keys()'.